storage/src/main/java/tech/pegasys/teku/storage/client/CombinedChainDataClient.java

```diff
@@ -149,11 +147,6 @@ public CombinedChainDataClient(
     return getBlockByBlockRoot(get_block_root_at_slot(state, slot));
   }
 
-  private boolean isHistoricalData(final UnsignedLong slot) {
-    final boolean finalizedPastFirstEpoch = !recentChainData.getFinalizedEpoch().equals(ZERO);
```

Comment:
I don't see why we need a special case for the first epoch / genesis block, so I cut this.  But if there's a valid reason, I can add this back in.

Reply:
I believe it was added because we were always going to disk to retrieve the genesis state due to it being inherently finalised.  But I think we've fixed `isFinalized` now so the last finalized slot is expected to be in memory.

Basically as long as the genesis state is retrieved from the memory store instead of disk it's fine to cut it.

Reply:
Ok, so basically we want to retrieve latest finalized data from memory, whether or not its on the epoch boundary?  And this applies for all finalized data, not just genesis?